TombEngine/Game/effects/effects.cpp

```diff
@@ -1749,3 +1523,115 @@ void TriggerMetalSparks(int x, int y, int z, int xv, int yv, int zv, int additio
 		}
 	}
 }
+
+void ProcessEffects(ItemInfo* item)
+{
+	constexpr auto MAX_LIGHT_FALLOFF = 13;
+	constexpr auto BURN_HEALTH_LARA = 7;
+	constexpr auto BURN_HEALTH_NPC = 1;
+	constexpr auto BURN_AFTERMATH_TIMEOUT = 2 * FPS;
+	constexpr auto BURN_DAMAGE_PROBABILITY = 1 / 8.0f;
+
+	if (item->Effect.Type == EffectType::None)
+		return;
+
+	if (item->Effect.Count > 0)
+	{
+		item->Effect.Count--;
+
+		if (!item->Effect.Count)
+		{
+			if (item->Effect.Type == EffectType::Burn)
+			{
+				item->Effect.Type = EffectType::Smoke;
+				item->Effect.Count = BURN_AFTERMATH_TIMEOUT;
+			}
+			else
+			{
+				item->Effect.Type = EffectType::None;
+				return;
+			}
+		}
+	}
+
+	int numMeshes = Objects[item->ObjectNumber].nmeshes;
+	for (int i = 0; i < numMeshes; i++)
+	{
+		auto pos = GetJointPosition(item, i);
+
+		switch (item->Effect.Type)
+		{
+		case EffectType::Burn:
+			if (TestProbability(1 / 8.0f))
+				TriggerFireFlame(pos.x, pos.y, pos.z, TestProbability(1 / 10.0f) ? FlameType::Trail : FlameType::Medium);
+			break;
+
+		case EffectType::Electric:
+			if (TestProbability(1 / 10.0f))
+				TriggerElectricSpark(&GameVector(pos.x, pos.y, pos.z, item->RoomNumber),
+					EulerAngles(0, GenerateInt(ANGLE(0), ANGLE(359)), 0), 2);
```

Comment:
For random angle generation, I have set up the function `Random::GenerateAngle()`. It's merely transitory for now but it would be nice to have it in place here for future developments.

Reply:
Ohh, thanks! This will surely come in handy.